nexus/src/app/sagas/instance_create.rs

```diff
@@ -1109,8 +1110,12 @@ async fn sic_set_boot_disk_undo(
 
     // If there was a boot disk, clear it. If there was not a boot disk,
     // this is a no-op.
+    if params.create_params.boot_disk.is_none() {
+        return Ok(());
+    }
```

Comment:
Hmm...it occurs to me that it might be nicer to just change the `make_saga_dag` function to just not append a `sic_set_boot_disk` node if there is no boot disk param. Like, change this: https://github.com/oxidecomputer/omicron/blob/41d2e2fafd77ac5dd5a17d060c25548276e5c8d1/nexus/src/app/sagas/instance_create.rs#L309

to:

```rust
if params.create_params.boot_disk.is_some() {
    builder.append(set_boot_disk_action());
}
```

That way, we would never do the forward _or_ reverse action if there's no boot disk.

Reply:
... huh, i suppose that makes sense. i'd read something in one of the comments here about Steno not handling dynamically-shaped sagas, but the comment earlier about `MAX_NICS_PER_INSTANCE` makes me think i'd read something old without realizing it.

Reply:
Hm...a saga's DAG can definitely depend on its _params_, so it's possible that the comment either meant something else by "dynamically-shaped", or was just out of date...